src/plugins/intel_gna/src/ops/util/util.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "util.hpp"
+
+using namespace GNAPluginNS::GNALimitations;
+using namespace ov::intel_gna;
+
+namespace {
+std::ostream& operator<<(std::ostream& os, const std::set<ov::element::Type>& t) {
+    for (auto it = t.begin(); it != t.end(); ++it) {
+        if (it != t.begin()) {
+            os << ", " << *it;
+        } else {
+            os << *it;
+        }
+    }
+    return os;
+}
+
+bool is_precision_supported(const std::shared_ptr<ov::Node>& node,
+                            const std::set<ov::element::Type>& supported_types,
+                            bool is_exception_allowed) {
+    if (!node) {
+        if (is_exception_allowed) {
+            THROW_GNA_EXCEPTION << "Node is empty!\n";
+        }
```

Comment:
maybe we could log error in case exceptions are not supported?

Reply:
> maybe we could log error in case exceptions are not supported?

Added assert here, because there is no sense to hide such kind of error in both situations